src/main/java/seedu/address/logic/commands/ClearCommand.java

```diff
@@ -12,12 +12,24 @@ public class ClearCommand extends Command {
 
     public static final String COMMAND_WORD = "clear";
     public static final String MESSAGE_SUCCESS = "Address book has been cleared!";
+    public static final String MESSAGE_CLEAR_CONFIRMATION = "This will permanently clear all contacts. "
+            + "Are you sure you want to execute this command? (y/n)";
 
+    private static final boolean requiresConfirmation = true;
 
     @Override
-    public CommandResult execute(Model model) {
+    protected CommandResult execute(Model model) {
         requireNonNull(model);
         model.setAddressBook(new AddressBook());
         return new CommandResult(MESSAGE_SUCCESS);
     }
+
+    @Override
+    public CommandResult execute(Model model, Boolean confirmationReceived) {
+        if (confirmationReceived.equals(requiresConfirmation)) {
+            return this.execute(model);
+        }
+        return new CommandResult(MESSAGE_CLEAR_CONFIRMATION, false, false, true);
+    }
+
```

Comment:
Perhaps, just leave it as `confirmationReceived.equals(true)`

Reply:
I mean if requiresConfirmation is false, then this.execute(model) should happen as well

Reply:
i see, then no issues with this.